Netkan/Sources/Github/GithubReleaseAsset.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.Collections.Generic;
+using CKAN.Versioning;
+
+namespace CKAN.NetKAN.Sources.Github
+{
+    public sealed class GithubReleaseAsset
+    {
+        public String        Name         { get; }
+        public ModuleVersion Version      { get; set; }
```

Comment:
> I'm not entirely happy about the `json.SafeAdd("version", ghAsset.Version?.ToString() ?? ghRelease.Tag.ToString());` line, ideas welcome.

My suggestion is to remove the `Version` property from the asset class, since it isn't an integral part of the upstream data. Then add a `Version` parameter to `GithubTransformer.TransformOne`; each of the two calls to `TransformOne` in `Transform` would know which value to use, either `new ModuleVersion(extractedVersion.Value)` or `rel.Tag.ToString()`.

Reply:
That sounds good, we'd get rid of two ugly-ish places of code at the same time.

Reply:
Turns out this also fixed another bug (at least in the test) we didn't see yet.

I initially changed the expected value of the second assertion in `Transform_SkipReleases_SkipsCorrectly`, because it returned the version from the filename.
However it shouldn't have done that, since its kref didn't have the new `version_from_asset`.

Turns out, the objects returned by the `apiMockUp` don't get regenerated for every test, but are permanent. So with `Transform_MultipleAssets_TransformsAll` running beforehand, and through it `Transform()` writing to `asset.Version`, it wasn't `null` anymore and that `json.SafeAdd()` line used it instead of the `ghRelease.Tag`.

Reminder to myself that I shouldn't just "fix" tests without thinking about why they are failing...